Tests/DatadogTests/Datadog/URLSessionAutoInstrumentation/Interception/AppStateListenerTests.swift

```diff
@@ -8,106 +8,106 @@ import XCTest
 @testable import Datadog
 
 class AppStateHistoryTests: XCTestCase {
-    func testForegroundDurationWithoutChanges() {
-        let startDate = Date.mockDecember15th2019At10AMUTC()
-        let history = AppStateHistory(
-            initialState: .init(isActive: true, date: startDate),
-            changes: [],
-            recentDate: startDate + 1.0
-        )
-
-        XCTAssertEqual(history.foregroundDuration, 1.0)
+    func testItBuildsAppStateFromUIApplicationState() {
+        XCTAssertEqual(AppState(uiApplicationState: .active), .active)
+        XCTAssertEqual(AppState(uiApplicationState: .inactive), .inactive)
+        XCTAssertEqual(AppState(uiApplicationState: .background), .background)
     }
 
-    func testForegroundDuration() {
-        let startDate = Date.mockDecember15th2019At10AMUTC()
-        let history = AppStateHistory(
-            initialState: .init(isActive: true, date: startDate),
-            changes: [
-                .init(isActive: false, date: startDate + 1.0),
-                .init(isActive: true, date: startDate + 2.0),
-                .init(isActive: false, date: startDate + 3.0),
-                .init(isActive: true, date: startDate + 4.0)
-            ],
-            recentDate: startDate + 5.0
-        )
+    func testWhenAppTransitionsBetweenForegroundAndBackground_itComputesTotalForegroundDuration() {
+        let numberOfForegroundSnapshots: Int = .mockRandom(min: 0, max: 20)
+        let numberOfBackgroundSnapshots: Int = .mockRandom(min: numberOfForegroundSnapshots == 0 ? 1 : 0, max: 20) // must include at least one snapshot
```

Comment:
Why not just
```suggestion
        let numberOfBackgroundSnapshots: Int = .mockRandom(min: 1, max: 20) // must include at least one snapshot
```

Reply:
I wanted to also cover scenario where `numberOfBackgroundSnapshots` is `0`. And this couldn't be done if `numberOfForegroundSnapshots` is `0` too, because we need at least one snapshot for:
```swift
initialSnapshot: allSnapshots[0],
```
In general, in this fuzzy test I wanted to cover all possibilities:
* app only in foreground,
* app only in background,
* app in both foreground and background.